test/serializers/attributes_test.rb

```diff
@@ -18,11 +18,31 @@ def test_attributes_definition
           @profile_serializer.class._attributes)
       end
 
+      def test_attributes
+        assert_equal({ name: 'Name 1', description: 'Description 1' },
+          @profile_serializer.attributes)
+      end
+
       def test_attributes_with_fields_option
         assert_equal({ name: 'Name 1' },
           @profile_serializer.attributes(fields: [:name]))
       end
 
+      def test_attributes_with_fields_method
+        @profile_serializer.define_singleton_method(:fields) do |keys|
+          keys - [:name]
+        end
+        assert_equal({ description: 'Description 1' },
+          @profile_serializer.attributes)
+      end
+
+      def test_attributes_with_fields_option_and_method
+        @profile_serializer.define_singleton_method(:fields) do |keys|
```

Comment:
I just mean that your test isn't testing https://github.com/rails-api/active_model_serializers/pull/1141/files#diff-5624f9cf35be4dc254412d979fd293c4R129 or https://github.com/rails-api/active_model_serializers/pull/1141/files#diff-5624f9cf35be4dc254412d979fd293c4R140 but is testing only https://github.com/rails-api/active_model_serializers/pull/1141/files#diff-be102247b73584d61072b4ac379b6e5fR32

But I see what you're saying. I'm going to leave it up to someone else to decide whether or not this is ready for merge.